src/Common/precompiled.h

```diff
@@ -210,11 +220,26 @@ typedef union _LARGE_INTEGER {
 
 // macros
 #if BOOST_OS_WINDOWS
-#define DLLEXPORT __declspec(dllexport)
+    #define DLLEXPORT __declspec(dllexport)
+    #define DLLIMPORT  __declspec(dllimport)
+    #define DEBUG_BREAK __debugbreak()
+    #define ALIGN(N) __declspec(align(N))
+    #define NOINLINE __declspec(noinline)
+    #define ASSUME(X) __assume((X)
+    #define THREAD_LOCAL __declspec(thread)
```

Comment:
[`thread_local`](https://en.cppreference.com/w/cpp/keyword/thread_local) is standard since C++11, you can drop this macro

Reply:
https://stackoverflow.com/questions/13106049/what-is-the-performance-penalty-of-c11-thread-local-variables-in-gcc-4-8

Reply:
Oh, wow, didn't know that. But it also says:

> This wrapper is not needed for in every use case of thread_local though. This can be revealed from decl2.c. The wrapper is generated only when:
> 
>     It is not function-local, and,
>         It is extern (the example shown above), or
>         The type has a non-trivial destructor (which is not allowed for __thread variables), or
>         The type variable is initialized by a non-constant-expression (which is also not allowed for __thread variables).
> 
> In all other use cases, it behaves the same as __thread. That means, unless you have some extern __thread variables, you could replace all __thread by thread_local without any loss of performance.

Reply:
I was more concerned about breaking windows versions because I have no easy way of testing. However, MSDN advises using thread_local so I expect it would be fine.

Reply:
As mentioned in the previous comment using `thread_local` the way it is currently used has no performance impact on GCC, while MSVC's documentation explicitly states that:

> On Windows, `thread_local` is functionally equivalent to the Microsoft-specific `__declspec( thread )` attribute.

See this for details: https://docs.microsoft.com/en-us/cpp/cpp/storage-classes-cpp?view=msvc-170#thread_local

Reply:
Thanks. https://github.com/cemu-project/Cemu/pull/101